examples/interpolate1d/CMakeLists.txt

```diff
@@ -5,12 +5,10 @@
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 if(HPX_WITH_EXAMPLES_HDF5)
-  if(NOT HDF5_CXX_LIBRARIES OR NOT HDF5_IS_PARALLEL)
-    hpx_error("The Interpolate1D example requires HDF C++ library enabled for thread-safety")
```

Comment:
What's the rationale of lifting this requirement?

Reply:
@hkaiser are you sure the `HDF5_IS_PARALLEL` cmake option and the `H5_HAVE_THREADSAFE` macro have the same purpose ?

Reply:
I've put it as it was before and I will update the packages variables when I have time cause some of them are deprecated, not only for HDF5